pandas/io/stata.py

```diff
@@ -734,6 +749,46 @@ def generate_value_label(self, byteorder: str) -> bytes:
         return bio.getvalue()
 
 
+class StataNonCatValueLabel(StataValueLabel):
+    """
+    Prepare formatted version of value labels
+
+    Parameters
+    ----------
+    labname : str
+        Value label name
+    value_labels: Dictionary
+        Mapping of values to labels
```

Comment:
might be easier to add a method in the super class to avoid overly defining this (e.g. you are repeating a bunch). is the signature the same? then you don't need an extended doc-string

Reply:
I wasn't sure how to best deal with the duplication, the only difference between the existing StataValueLabel and new non-categorical value label is the init: the non-categorical takes the label name and a dict of value labels directly, the existing `StataValueLabel` is created from a categorical series and pulls the name and labels from it.

The two options I've thought of for getting rid of the duplication are:
* change `StataValueLabel` to init from a label name and value_labels dictionary then add a `from_cat` method to extract the name and label from a Categorical
* add labname and value_labels to the StataValueLabel init and let it be created from either those or the catarray

Both would get rid of the new non categorical label class entirely, the first would be a breaking change to StataValueLabel. Would one of these be acceptable or is there a better way that I'm not thinking of?

Reply:
@jreback I moved the duplicated attribute definitions to _prepare_value_labels. Would either of the changes from above be better?